plugin/plaintextmodel.cpp

```diff
@@ -280,6 +309,10 @@ void PlainTextModel::Reader::run()
         QCoreApplication::postEvent(this, event);
     } else {
         QTextStream stream(&file);
+        QByteArray ba = file.readAll();
+        QString name = detect(ba.data(), ba.size())->name();
+        stream.setCodec(QTextCodec::codecForName( name.toStdString().data() ));
+        stream.seek(0);
```

Comment:
Detection done twice?

Reply:
What do you mean?

Reply:
There are two places where detect() is called.

Reply:
Done